execute/plugin.go

```diff
@@ -584,18 +588,14 @@ func (p *Plugin) Reports(
 		return nil, nil
 	}
 
-	// TODO: Handle multiple reports.
-	encodedReport, err := p.reportCodec.Encode(ctx, decodedOutcome.Reports[0])
+	reportInfos, err := extractReportInfo(decodedOutcome)
+	lggr.Debugw("report info in UnfinalizedReports()", "reportInfos", reportInfos)
```

Comment:
This log should be added after the error check. Also what is "UnfinalizedReports()"?